src/PostgREST/Error.hs

```diff
@@ -633,3 +684,47 @@ buildErrorCode code = "PGRST" <> case code of
   JWTErrorCode02         -> "302"
 
   InternalErrorCode00    -> "X00"
+
+-- For parsing byteString to JSON Object, used for allowing full response control
+data PGRSTRaiseErrMessage = PGRSTRaiseErrMessage {
+  getCode    :: Text,
+  getMessage :: Text,
+  getDetails :: Text,
+  getHint    :: Text
+}
+
+data PGRSTRaiseErrDetails = PGRSTRaiseErrDetails {
+  getStatus  :: Int,
+  getHeaders :: Map Text Text
+}
+
+instance JSON.FromJSON PGRSTRaiseErrMessage where
+  parseJSON (JSON.Object m) =
+    PGRSTRaiseErrMessage
+      <$> m .: "code"
+      <*> m .: "message"
+      <*> m .:? "details" .!= ""
+      <*> m .:? "hint"    .!= ""
+
+  parseJSON _ = mzero
+
+instance JSON.FromJSON PGRSTRaiseErrDetails where
+  parseJSON (JSON.Object d) =
+    PGRSTRaiseErrDetails
+      <$> d .: "status"
+      <*> d .: "headers"
+
+  parseJSON _ = mzero
+
+parseMessage :: ByteString -> Maybe PGRSTRaiseErrMessage
+parseMessage m =
+  case ATP.parse JSON.json m of
+    (ATP.Done _ r) -> JSON.parseMaybe JSON.parseJSON r :: Maybe PGRSTRaiseErrMessage
+    _          -> Nothing
```

Comment:
I think this can be done directly with Aeson using `decode` or [`decodeStrict`](https://hackage.haskell.org/package/aeson-2.2.0.0/docs/Data-Aeson.html#v:decodeStrict) for this type of ByteString?

```suggestion
parseMessage :: ByteString -> Maybe PGRSTRaiseErrMessage
parseMessage m = JSON.decodeStrict m :: Maybe PGRSTRaiseErrMessage
```

Reply:
@laurenceisla Thanks! Now we can also remove `attoparsec`.